gsa/src/web/pages/portlists/component.js

```diff
@@ -25,186 +25,205 @@ import {isDefined} from 'gmp/utils/identity';
 import {shorten} from 'gmp/utils/string';
 
 import PropTypes from 'web/utils/proptypes';
-import withGmp from 'web/utils/withGmp';
+import useGmp from 'web/utils/useGmp';
+import reducer from 'web/utils/stateReducer';
 
 import EntityComponent from 'web/entity/component';
 
 import ImportPortListDialog from './importdialog';
 import PortListsDialog from './dialog';
 import PortRangeDialog from './portrangedialog';
 
-class PortListComponent extends React.Component {
-  constructor(...args) {
-    super(...args);
-
-    this.state = {
-      importDialogVisible: false,
-      portListDialogVisible: false,
-      portRangeDialogVisible: false,
-    };
-
-    this.created_port_ranges = [];
-    this.deleted_port_ranges = [];
-
-    this.handleCloseImportDialog = this.handleCloseImportDialog.bind(this);
-    this.handleClosePortListDialog = this.handleClosePortListDialog.bind(this);
-    this.handleCloseNewPortRangeDialog = this.handleCloseNewPortRangeDialog.bind(
-      this,
-    );
-    this.openImportDialog = this.openImportDialog.bind(this);
-    this.openNewPortRangeDialog = this.openNewPortRangeDialog.bind(this);
-    this.openPortListDialog = this.openPortListDialog.bind(this);
-    this.handleDeletePortRange = this.handleDeletePortRange.bind(this);
-    this.handleImportPortList = this.handleImportPortList.bind(this);
-    this.handleSavePortList = this.handleSavePortList.bind(this);
-    this.handleSavePortRange = this.handleSavePortRange.bind(this);
-    this.handleTmpAddPortRange = this.handleTmpAddPortRange.bind(this);
-    this.handleTmpDeletePortRange = this.handleTmpDeletePortRange.bind(this);
-  }
-
-  openPortListDialog(entity) {
-    const {gmp} = this.props;
-
+const PortListComponent = ({
+  children,
+  onCloned,
+  onCloneError,
+  onCreated,
+  onCreateError,
+  onDeleted,
+  onDeleteError,
+  onDownloaded,
+  onDownloadError,
+  onImported,
+  onImportError,
+  onInteraction,
+  onSaved,
+  onSaveError,
+}) => {
+  const gmp = useGmp();
+
+  const [state, dispatch] = useReducer(reducer, {
+    importDialogVisible: false,
+    portListDialogVisible: false,
+    portRangeDialogVisible: false,
+  });
+
+  const [createdPortRanges, setCreatedPortRanges] = useState([]);
```

Comment:
Is special treatment here unnecessary? Can they be used like normal states?

Reply:
Just some background for using `this.foo` for a *global* variable in the past. Sometimes it wasn't necessary to re-render if some variable has been set. E.g. it wasn't displayed after some loading from the backend. Therefore using a global variable instead of updating the state avoid the re-render. I am not sure if this is still the case therefore don't think about re-rendering anymore (at the moment) and don't look back.